src/librustc_lint/types.rs

```diff
@@ -658,6 +659,15 @@ impl<'a, 'tcx> ImproperCTypesVisitor<'a, 'tcx> {
                                 ParamEnv::reveal_all(),
                                 field.ty(cx, substs),
                             );
+                            if def.repr.transparent() {
+                                let is_zst = cx
```

Comment:
It'd be helpful to include the corresponding comment about why ZST fields can be skipped.